command.py

```diff
@@ -1,24 +1,70 @@
+#!/usr/bin/env python3
+
 import argparse
-import os
+import logging
+import pathlib
 import pkgutil
+import os
 
-if __name__ == '__main__':
-
-    COMMANDS_DIRECTORY = 'commands'
+import coloredlogs
+import verboselogs
 
-    parser = argparse.ArgumentParser(description='Managing %s commands' % os.environ.get("PROJECT_NAME"))
-    subparsers = parser.add_subparsers(help='sub commands')
 
-    subcommands = []
-    for _, command_name, _ in pkgutil.iter_modules([COMMANDS_DIRECTORY]):
-        path = "%s.%s" % (COMMANDS_DIRECTORY, command_name)
-        commands_import = __import__(path)
-        subcommands.append(getattr(commands_import, command_name))
+def main(project_name: str):
+    # Discover all the commands defined in derived image
+    parser = argparse.ArgumentParser(description=f"Managing {project_name} commands")
+    parser.add_argument(
+        "--log-level",
+        choices=sorted(logging._nameToLevel, key=lambda name: logging._nameToLevel[name]),
+        default="INFO",
+    )
 
-    for subcommand in subcommands:
-        subcommand.parse_args(subparsers)
+    # Import our commands and ask them to populate the subparsers with their argument configurations
+    gather_commands(
+        subparsers=parser.add_subparsers(help="Sub-commands"),
+        commands_directory=pathlib.Path(os.environ.get("ALADDIN_COMMANDS_PATH", "/code/commands")),
+    )
 
+    # Actually parse the arguments
     args = parser.parse_args()
 
+    # Use the provided log level and upgrade our logging capabilities
+    coloredlogs.install(
```

Comment:
This is neat.  That said, are we following this pattern anywhere else in aladdin?

Reply:
<ominous music and startling thunderclap> not yet... ;)

aladdin does use something called `colorlog`, but i think these options are more preferable